src/js/components/modals/AppModalComponent.jsx

```diff
@@ -165,11 +175,15 @@ var AppModalComponent = React.createClass({
 
     // Ports should always be an Array.
     if ("ports" in modelAttrs) {
-      var portStrings = modelAttrs.ports.split(",");
-      modelAttrs.ports = portStrings.map(function (p) {
-        var port = parseInt(p, 10);
-        return isNaN(port) ? p : port;
-      });
+      if (modelAttrs.ports === "") {
```

Comment:
FTR here's the refactoring issue https://github.com/mesosphere/marathon/issues/2105